service/controller/chartconfig/v7/resource/chartstatus/create.go

```diff
@@ -29,30 +31,48 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	if key.ReleaseStatus(customObject) != releaseContent.Status {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting status for release '%s' status to '%s'", releaseName, releaseContent.Status))
-
-		customObjectCopy := customObject.DeepCopy()
-		customObjectCopy.Status.ReleaseStatus = releaseContent.Status
+	var status, reason string
+	{
+		if key.IsCordoned(customObject) {
+			status = releaseStatusCordoned
+			reason = key.CordonReason(customObject)
+		} else {
+			status = releaseContent.Status
+			if releaseContent.Status != releaseStatusDeployed {
+				releaseHistory, err := r.helmClient.GetReleaseHistory(ctx, releaseName)
+				if err != nil {
+					return microerror.Mask(err)
+				}
 
-		if releaseContent.Status != releaseStatusDeployed {
-			releaseHistory, err := r.helmClient.GetReleaseHistory(ctx, releaseName)
-			if err != nil {
-				return microerror.Mask(err)
+				reason = releaseHistory.Description
 			}
+		}
+	}
 
-			customObjectCopy.Status.Reason = releaseHistory.Description
-		} else {
-			customObjectCopy.Status.Reason = ""
+	chartConfigStatus := v1alpha1.ChartConfigStatus{
+		ReleaseStatus: status,
+		Reason:        reason,
+	}
+
+	if customObject.Status != chartConfigStatus {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting status for release '%s' status to '%s'", releaseName, status))
+
+		// Get chartconfig CR again to ensure the resource version is correct.
+		currentCR, err := r.g8sClient.CoreV1alpha1().ChartConfigs(customObject.Namespace).Get(customObject.Name, metav1.GetOptions{})
+		if err != nil {
+			return microerror.Mask(err)
 		}
-		_, err := r.g8sClient.CoreV1alpha1().ChartConfigs(customObject.Namespace).UpdateStatus(customObjectCopy)
+
+		currentCR.Status = chartConfigStatus
```

Comment:
I think we should only update the properties we change ourselves. You may just overwrite everything else now.

Reply:
We have two values in status, `reason`, `releaseStatus`. Do you think we have to split them over to update only variables that need to update?

Reply:
Thought experiment. Tomorrow we extend the status and next week nobody knows this anymore. Some new colleague joining next month tries to work with this and has to deal with problems we introduced now. Why should we do this?

Reply:
Not sure that we would need to add up more variables in status, but by your suggestion does it looks like below, right?
```
		if currentCR.Status.Reason != chartConfigStatus.Reason {
			# loging reason update
			currentCR.Status.Reason = chartConfigStatus.Reason
		}

		if currentCR.Status.ReleaseStatus != chartConfigStatus.ReleaseStatus {
			# logging status update
			currentCR.Status.ReleaseStatus = chartConfigStatus.ReleaseStatus
		}
```

Reply:
Something like this, yes. 👍

Reply:
@xh3b4sd I remove original logs to make it separate logs, PTAL when you have time!